src/basilisp/lang/reader.py

```diff
@@ -1217,9 +1221,10 @@ def identify_and_replace(f):
     if len(numbered_args) > 0:
         max_arg = max(numbered_args)
         arg_list = [sym_replacement(str(i)) for i in range(1, max_arg + 1)]
-        if "rest" in arg_set:
-            arg_list.append(_AMPERSAND)
-            arg_list.append(sym_replacement("rest"))
+
+    if "rest" in arg_set:
+        arg_list.append(_AMPERSAND)
+        arg_list.append(sym_replacement("rest"))
```

Comment:
This was a separate bug where `#(vec %&)` forms didn't expand correctly with or without syntax quotes to capture the rest argument.